objectstorage/cached_object.go

```diff
@@ -319,37 +319,65 @@ func (cachedObject *CachedObjectImpl) BatchWrite(batchedMuts kvstore.BatchedMuta
 		return
 	}
 
-	if storableObject.ShouldPersist() && storableObject.IsModified() {
+	if !storableObject.ShouldPersist() {
+		// set object as not modified even if we do not persist because it otherwise will not get evicted from the cache
 		storableObject.SetModified(false)
+		return
+	}
 
-		var marshaledValue []byte
-		if !objectStorage.options.keysOnly {
-			marshaledValue = storableObject.ObjectStorageValue()
-		}
+	if !storableObject.IsModified() {
```

Comment:
Why is this check not done before the other one?
Can't we always return if the object is not modified, even if we don't persist?